dev/DynamicDependency/MddLifetimeManagement.cpp

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "MddLifetimeManagement.h"
+#include "MddLifetimeManagementTest.h"
+
+static std::wstring g_test_ddlmPackageNamePrefix;
+static std::wstring g_test_ddlmPackagePublisherId;
+
+STDAPI MddLifetimeManagementTestInitialize(
+    _In_ PCWSTR ddlmPackageNamePrefix,
+    _In_ PCWSTR ddlPackagePublisherId) noexcept try
+{
+    RETURN_HR_IF(E_INVALIDARG, !ddlmPackageNamePrefix);
+    RETURN_HR_IF(E_INVALIDARG, *ddlmPackageNamePrefix == L'0');
+    RETURN_HR_IF(E_INVALIDARG, !ddlPackagePublisherId);
+    RETURN_HR_IF(E_INVALIDARG, *ddlPackagePublisherId == L'0');
+
+    g_test_ddlmPackageNamePrefix = ddlmPackageNamePrefix;
+    g_test_ddlmPackagePublisherId = ddlPackagePublisherId;
+    return S_OK;
+} CATCH_RETURN();
+
+namespace MddCore::LifetimeManagement
+{
+class DDLMPackage
+{
+public:
+    DDLMPackage(
+        const std::wstring packageFullName,
+        PACKAGE_VERSION version) :
+        m_packageFullName(packageFullName),
+        m_version(version)
+    {
+    }
+
+    constexpr bool operator>(const DDLMPackage& rhs) const
+    {
+        return m_version.Version > rhs.m_version.Version;
+    }
+
+    const std::wstring& PackageFullName() const
+    {
+        return m_packageFullName;
+    }
+
+    void RemovePackage(
+        winrt::Windows::Management::Deployment::PackageManager& packageManager) const
+    {
+        try
+        {
+            auto deploymentResult{ packageManager.RemovePackageAsync(m_packageFullName.c_str()).get() };
+            if (!deploymentResult)
+            {
+                const auto hr{ deploymentResult.ExtendedErrorCode() };
+                if (hr == HRESULT_FROM_WIN32(ERROR_PACKAGES_IN_USE))
+                {
+                    (void) LOG_HR_MSG(deploymentResult.ExtendedErrorCode(), "RemovePackage('%ls') = 0x%0X %ls. Will try again later", m_packageFullName.c_str(), deploymentResult.ExtendedErrorCode(), deploymentResult.ErrorText().c_str());
+                }
+                else
+                {
+                    (void) LOG_HR_MSG(deploymentResult.ExtendedErrorCode(), "RemovePackage('%ls') = 0x%0X %ls", m_packageFullName.c_str(), deploymentResult.ExtendedErrorCode(), deploymentResult.ErrorText().c_str());
+                }
+            }
+        }
+        catch (...)
```

Comment:
WIL has `CATCH_LOG()` as well

Reply:
Meant to include packageFullName in the log else yes, CATCH_LOG would suffice. Fixec